graph-manager-docs/source/federation.mdx

```diff
@@ -123,9 +123,115 @@ server.listen().then(({ url }) => {
+to other modifications. This will help avoid the situation that the query planner generates queries that fail validation in downstream services or violate
+resolvers' expectations. Some examples of this include:
+
+1. Modifying `@key` or `@requires` dependencies while simultaneously modifying dependency fields
+2. Removing a type implementation from an interface
+
+For example, with an interface that has implementations in multiple services, such as the following:
+
+```
+// Channel service
+interface Channel @key(fields: "id") {
+  id: ID!
+}
+
+// Web service
+type WebChannel implements Channel @key(fields: "id") {
+  id: ID!
+}
+
+// Email service
+type EmailChannel implements Channel @key(fields: "id") {
+  id: ID!
+}
+```
+
+In order to remove the `EmailChannel` implementation of the `Channel` interface, you'll want to perform the following steps:
+
+1. Register a new `email` service that removes the `EmailChannel` implementation of `Channel`
+1. Deploy a new Email service that removes the implementation
+
+The first step will cause the query-planner to stop sending along fragments `...on EmailChannel`, which would fail validation if sent
+to a service that did not have awareness of that type!
+
+#### Modifying query-planning logic
+
+You'll want to treat migrations of query planning logic similarly to how you may treat database migrations, by carefully considering the effects on the
+downstream services as the query planner changes, and planning for "double reading" as appropriate. For example, consider the case of migrating a
+`@required` field in an extension from one to another. In this example, we'll consider the `Products` service and the `Reviews` service, like so:
+
+```js
+// Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String!
+}
+
+// Review Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  reviews: [Review]! @requires(fields: "nameLowercase")
+  nameLowercase: String! @external
+}
+```
+
+If the `nameLowercase` field is to be deprecated and replaced with the `name` field
+
+```js
+// Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String! @deprecated
+  name: String!
+}
+
+// Reviews Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  nameLowercase: String! @external
+  name: String! @external
+  reviews: [Review]! @requires(fields: "name")
+}
+```
+
+To do this migration in-place, you would perform the following steps:
+
+1. Modify the `Products` service to add the new field (as per usual, first deploy all replicas, then `apollo service:push` the new partial schema)
+1. Deploy a new version of the `Reviews` service with a resolver that accepts _either_ `nameLowercase` or `name` in the source object
+1. Modify the schema of the `Reviews` service in the registry to update query-planning logic to `@require` the new `name` field
```

Comment:
`@require` isn't the directive, right? It's `@requires`? Then rephrase to:

"Modify the `Reviews` service's schema in the registry so that it `@requires` the new `name` field"

(The pattern of using the directive as a word in the sentence is fine, but we shouldn't alter _what_ that directive is)

Reply:
sg